update.go

```
@@ -30,366 +30,29 @@
-		}
-		//self, err := osext.Executable()
-		//if err != nil {
-		// fail update, couldn't figure out path to self
-		//return
-		//}
-		// TODO(bgentry): logger isn't on Windows. Replace w/ proper error reports.
-		if err := u.update(); err != nil {
-			return err
-		}
-	}
-	return nil
-}
-
-func (u *Updater) wantUpdate() bool {
-	if strings.Contains(u.CurrentVersion, "dev") {
-		return false
-	} else {
-		return true
-	}
-}
-
-func (u *Updater) update() error {
-	path, err := osext.Executable()
-	if err != nil {
-		return err
-	}
-	old, err := os.Open(path)
-	if err != nil {
-		return err
-	}
-	defer old.Close()
-
-	err = u.fetchInfo()
-	if err != nil {
-		log.Println(err)
-		return err
-	}
-	if u.Info.Version == u.CurrentVersion {
-		return nil
-	}
-	bin, err := u.fetchAndVerifyPatch(old)
-	if err != nil {
-		if err == ErrHashMismatch {
-			log.Println("update: hash mismatch from patched binary")
-		} else {
-			if u.DiffURL != "" {
-				log.Println("update: patching binary,", err)
-			}
-		}
-
-		bin, err = u.fetchAndVerifyFullBin()
-		if err != nil {
-			if err == ErrHashMismatch {
-				log.Println("update: hash mismatch from full binary")
-			} else {
-				log.Println("update: fetching full binary,", err)
-			}
-			return err
-		}
-	}
-
-	// close the old binary before installing because on windows
-	// it can't be renamed if a handle to the file is still open
-	old.Close()
-
-	err, errRecover := up.FromStream(bytes.NewBuffer(bin))
-	if errRecover != nil {
-		log.Errorf("update and recovery errors: %q %q", err, errRecover)
-		return fmt.Errorf("update and recovery errors: %q %q", err, errRecover)
-	}
-	if err != nil {
-		return err
-	}
-
-	// remove config.ini so at restart the package will extract again
-	//shutil.CopyFile(*configIni, *configIni+".bak", false)
-	//os.Remove(*configIni)
-
-	log.Println("Restarting because update!")
 
+	c.JSON(200, gin.H{"success": "Please wait a moment while the agent reboots itself"})
+	path, _ := osext.Executable()
```

Please do not ignore errors but handle it and exit if fails.